pkg/util/limiter/per_query_limiter.go

```diff
@@ -0,0 +1,119 @@
+package limiter
+
+import (
+	"context"
+	"fmt"
+	"github.com/cortexproject/cortex/pkg/cortexpb"
+	"github.com/cortexproject/cortex/pkg/ingester/client"
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/cortexproject/cortex/pkg/util/validation"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"go.uber.org/atomic"
+	"sync"
+)
+
+type PerQueryLimiter struct {
```

Comment:
[nit] We can simplify the name and remove the `Per` prefix everywhere. `QueryLimiter` looks self-explanatory. WDYT?